### PR TITLE
[MODRR-9]-Added PUT endpoint for patron permission.

### PR DESCRIPTION
**Pull request by SinghAdes**
## Purpose
https://folio-org.atlassian.net/browse/MODRR-9

## Approach
Added PUT endpoint for PatronPermissions. 
#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

**Review by gurleenkaurbp — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -62,15 +66,15 @@
...
+          "pathPattern": "/reading-room-patron-permission/{patronId}",
           "permissionsRequired": [
             "reading-room-patron-permission.item.get"
           ],
           "modulePermissions": []
         },
         {
           "methods": ["PUT"],
-          "pathPattern": "/reading-room-patron-permission/users/{userId}",
+          "pathPattern": "/reading-room-patron-permission/{patronId}",
```
Comment (Vignesh-kalyanasundaram) [inline descriptors/ModuleDescriptor-template.json:77]: Please add the necessary module permission

File: src/main/java/org/folio/readingroom/service/impl/PatronPermissionsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import org.folio.readingroom.service.UserService;
+import org.folio.readingroom.service.converter.Mapper;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class PatronPermissionsServiceImpl implements PatronPermissionsService {
+  private final PatonPermissionsRepository patonPermissionsRepository;
+  private final Mapper readingRoomMapper;
```
Comment (Vignesh-kalyanasundaram) [inline src/main/java/org/folio/readingroom/service/impl/PatronPermissionsServiceImpl.java:20]: please change readingRoomMapper to mapper

File: src/test/java/org/folio/readingroom/controller/PatronPermissionControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    PatronPermission patronPermission = createPatronPermission(UUID.randomUUID(),
+      UUID.randomUUID(), UUID.fromString("2305005b-ca51-4a04-87fd-938eefa8f6df"));
+    this.mockMvc.perform(
+        put("/reading-room-patron-permission/2305005b-ca51-4a04-87fd-939eefa8f6df")
+          .content(asJsonString(patronPermission))
+          .headers(defaultHeaders())
+          .contentType(MediaType.APPLICATION_JSON)
+          .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is4xxClientError());
+  }
```
Comment (Vignesh-kalyanasundaram) [inline src/test/java/org/folio/readingroom/controller/PatronPermissionControllerTest.java:89]: we can also add one more error scenario
Try to update a patron permission with invalid reading room Id.

File: src/test/java/org/folio/readingroom/controller/PatronPermissionControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import org.folio.readingroom.repository.PatonPermissionsRepository;
+import org.folio.readingroom.repository.ReadingRoomRepository;
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+
+class PatronPermissionControllerTest extends BaseIT {
+
```
Comment (Vignesh-kalyanasundaram) [inline src/test/java/org/folio/readingroom/controller/PatronPermissionControllerTest.java:29]: I think the below fields are not used anywhere in this class. If it is not used, we can remove all these fields.

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**